samples/go/url-fetch/fetch.go

```diff
@@ -77,7 +95,10 @@ func main() {
 
 		r = resp.Body
 		contentLength = resp.ContentLength
-		sourceType, sourceVal = "url", url
+		additionalMetaInfo["url"] = url
```

Comment:
Here you could just use `root.Meta` from above suggestion.

Reply:
I just tried that out and found I'd have to blank out some fields. For example if you've already fetched a file then fetch a url, root.Meta.File will contain a value that needs to be removed.

Also spec.CreateCommitMetaStruct adds some other info such as the commit date which I'd have to replicate. 

Because of this I decided to leave the additionalMetaInfo map in place. Let me know if I'm misunderstanding.

Reply:
Makes sense. It's a shame it's not easier to use this pattern with the meta fields. We should go back through and clean up the API.